jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +38,33 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
+    self._evict_entries_if_necessary(key, value)
+    self._create_a_file(self._path, key, value)
+
+  def _evict_entries_if_necessary(self, key:str, value:bytes):
+    directory_size = self._get_cache_directory_size()
+    new_file_size = len(value)
+
+    assert (new_file_size <= self._max_cache_size_bytes), "This compilation file is larger than the size of the cache"
+
+    while new_file_size + directory_size > self._max_cache_size_bytes:
+      last_time = 0.0
+      file_to_delete = ""
```

Comment:
Style nit: use `None` instead of `""` to indicate that file_to_delete hasn't been set yet